include/seqan3/core/bit_manipulation.hpp

```diff
@@ -96,4 +105,48 @@ constexpr uint8_t bit_scan_reverse(unsigned_t n)
 #endif
 }
 
+/*!\brief Convert the byte encoding of integer values to little-endian byte order.
+ * \ingroup core
+ * \tparam type The type of the value to convert; must model std::Integral.
+ * \param  in   The input value to convert.
+ * \returns the converted value in little-endian byte-order.
+ *
+ * \details
+ *
+ * This function swaps the bytes if the host system uses big endian and if the size of the integral types is not
+ * bigger than 8 byte. Otherwise, the function does nothing and returns the unchanged input value.
```

Comment:
What about 1 byte integers? 
If this function cannot work on 1 byte or >8 byte integers than maybe it should be constrained to those instead of doing nothing?

And what about signed integers?

Reply:
Well, a 1 byte does not have a different byte-order on little endian vs big endian 😏. 
Perhaps "not bigger than 8 byte" is misleading, since the function is already constrained on integral types anyway. I'll remove the comment

Reply:
> Well, a 1 byte does not have a different byte-order on little endian vs big endian

Oh, I though little/big endian is about the **bit** order not the byte order.. 

> Perhaps "not bigger than 8 byte" is misleading, since the function is already constrained on integral types anyway.

does an GCC's __int128 not model std::Integral?

And what about signed integers?

Reply:
> does an GCC's __int128 not model std::Integral?

Ok, I did more reading. In general, there is more to it. Integral does not protect against possibly larger types for example `implementation-defined extended integer types`. In addition, the standard does only guarantee a minimum width of the standard integer types. Thus, there could be systems with ints large than 8 byte or 12 and 48 bit width integers (older processors). 
But, it is very unlikely that we will ever run on such systems.
I will constraint it to is_integer which only holds for a predefined set of types.

> And what about signed integers?

It works for signed and unsigned. The system merely interprets the same number in a different byte-order, which does has no affect on the value of the represented bits.